### PR TITLE
Demonstration of `description`

### PR DESCRIPTION
**Pull request by xeho91**
## Issues

1. JSDoc meta descriptions are lost with `lang="ts"`. https://github.com/storybookjs/addon-svelte-csf/issues/175 
1. Encountered issue with setting the description in the **Svelte CSF** format, above the `export const meta`.
3. In the `Svelte CSF` format, setting explicitly the description inside the `<Story />` component via `parameter` prop doesn't work, nor override the comment.
4. ~~In the current version of `@storybook/addon-svelte-csf` `<Story>` and `<Template />` component, it doesn't have typings for `tags` prop, so the TypeScript will complain - which is unrelated to the subject of this demonstration.~~
5. ~~Might need to wait for https://github.com/storybookjs/storybook/pull/26634 to be released, so we can improve the examples showcase for this case.~~

### CODE REVIEW

**Comment by xeho91** [comment]
FYI.

Regarding the issue `1.` and `2.`, while I was working on the experimental support for Svelte `v5` _(on `@storybook/addon-svelte-csf`)_, I have solved those issues by refactoring the parser.

File: src/docs/DescriptionExplicit.stories.svelte [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+			},
+		},
+	} satisfies Meta<DescriptionComponent>;
+</script>
+
+<Template let:args>
+	<DescriptionComponent {...args} />
+</Template>
+
+<Story name="From export meta" args={{ useMeta: true }} tags={['!dev']} />
```
Comment (JReinhold) [inline src/docs/DescriptionExplicit.stories.svelte:34]: I don't think this story is needed TBH, as it's already covered by "Not described"

File: src/docs/DescriptionExplicit.stories.svelte [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+
+<!-- This is a description for the story `HTML comment`, it was set as **HTML comment above the story**. -->
+<Story
+	name="Override HTML comment"
```
Comment (JReinhold) [inline src/docs/DescriptionExplicit.stories.svelte:38]:
```suggestion
	name="From story parameters"
```

File: src/docs/DescriptionExplicit.stories.ts [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+	},
+} satisfies Meta<DescriptionComponent>;
+
+export default meta;
+
+type Story = StoryObj<DescriptionComponent>;
+
+export const FromMeta: Story = { args: { useMeta: true }, tags: ["!dev"] };
+
+/** This is a description for the story `JS Doc`, it was set above `export const JDDoc`. */
```
Comment (JReinhold) [inline src/docs/DescriptionExplicit.stories.ts:27]:
```suggestion
/** This is a description for the story `JSDoc`, it was set above `export const`. */
```

File: src/docs/DescriptionExplicit.stories.ts [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+	parameters: {
+		controls: { disable: true },
+	},
+} satisfies Meta<DescriptionComponent>;
+
+export default meta;
+
+type Story = StoryObj<DescriptionComponent>;
+
+export const FromMeta: Story = { args: { useMeta: true }, tags: ["!dev"] };
```
Comment (JReinhold) [inline src/docs/DescriptionExplicit.stories.ts:25]: I think you can set `"!dev"` on the meta level for this to work for all stories, as they inherit that.

```suggestion
export const FromMeta: Story = { args: { useMeta: true }, tags: ["!dev"] };
```

File: src/docs/DescriptionFromComment.stories.svelte [hunk 5]
```diff
@@ -0,0 +1,33 @@
...
+	import { Story, Template } from '@storybook/addon-svelte-csf';
+	import type { Meta } from '@storybook/svelte';
+
+	import DescriptionComponent from './DescriptionComponent.svelte';
+
+	/**
+	 * This description was set above the `export const meta` inside the module tag `<script context="module">`.
+	 */
+	export const meta = {
+		title: 'Description From Comment/Svelte CSF',
```
Comment (JReinhold) [inline src/docs/DescriptionFromComment.stories.svelte:11]: Could you move all stories under a single "Description" title? Like `"Description/From Comment/Svelte CSF"`. I think that would help a bit with navigating the Storybook

File: src/docs/DescriptionFromComment.stories.ts [hunk 6]
```diff
@@ -0,0 +1,48 @@
+import type { Meta, StoryObj } from "@storybook/svelte";
+
+import DescriptionComponent from "./DescriptionComponent.svelte";
+
+/**
+ * This description was set above the `const meta` in the regular CSF file.
+ */
+const meta = {
+	title: "Description Set Explicitly/Regular CSF",
```
Comment (JReinhold) [inline src/docs/DescriptionFromComment.stories.ts:9]:
```suggestion
	title: "Description/From parameters/Regular CSF",
```

File: src/docs/DescriptionFromComment.stories.ts [hunk 7]
```diff
@@ -0,0 +1,48 @@
...
+} satisfies Meta<DescriptionComponent>;
+
+export default meta;
+
+type Story = StoryObj<DescriptionComponent>;
+
+export const FromMeta: Story = { args: { useMeta: true }, tags: ["!dev"] };
+
+/** This is a description for the story `JSDoc`, it was set above `export const JDDoc`. */
+export const JSDoc: Story = {
```
Comment (JReinhold) [inline src/docs/DescriptionFromComment.stories.ts:35]:
```suggestion
/** This is a description for the story `Parameters`, it was set above `export const Parameters`. */
export const Parameters: Story = {
```

File: src/docs/DescriptionFromComment.stories.ts [hunk 8]
```diff
@@ -0,0 +1,48 @@
...
+	args: { useStory: true },
+	parameters: {
+		docs: {
+			description: {
+				story:
+					"This explicitly set description overrides the JSDoc comment above the `export const JSDoc`.",
```
Comment (JReinhold) [inline src/docs/DescriptionFromComment.stories.ts:41]:
```suggestion
					"This explicitly set description overrides the JSDoc comment above the `export const Parameters`.",
```

**Review by JReinhold — APPROVED** [review]
Thank you! ❤️

Added some very minor comments

> Encountered issue with setting the description in the Svelte CSF format, above the export const meta

Could you elaborate on this?

> In the Svelte CSF format, setting explicitly the description inside the <Story /> component via parameter prop doesn't work, nor override the comment.

Great find! Sounds like a regular bug.

> In the current version of @storybook/addon-svelte-csf <Story> and <Template /> component, it doesn't have typings for tags prop, so the TypeScript will complain - which is unrelated to the subject of this demonstration.

Thanks!

> Might need to wait for https://github.com/storybookjs/storybook/pull/26634 to be released, so we can improve the examples showcase for this case.

This is already in the 8.1.0 beta which we're using in this repo, so it should be fine.

File: README.md [hunk 9]
```diff
@@ -25,7 +25,7 @@ These are the following Storybook features we attempt to uncover in this project
 | ⏳     | [`render`](https://storybook.js.org/docs/api/csf#custom-render-functions)                                                                                   |         |       |
 | ⏳     | [`play`](https://storybook.js.org/docs/writing-stories/play-function)                                                                                       |         |       |
 | ⏳     | [How to type the meta object and stories](https://storybook.js.org/docs/writing-stories/typescript)                                                         |         |       |
-| ⏳     | [JSDoc comments as descriptions](https://storybook.js.org/docs/api/doc-block-description#writing-descriptions) (from component, meta and stories)           |         |       |
+| 🟡     | [JSDoc comments as descriptions](https://storybook.js.org/docs/api/doc-block-description#writing-descriptions) (from component, meta and stories)           |         |       |
```
Comment (JReinhold) [inline README.md:28]: I changed the structure of this table slightly, to have a "Links" column.

Could you add links to the stories (using https://main--663faba8e103e55dccd640dc.chromatic.com as a base URL) and to this PR?

**Comment by xeho91** [comment]
> > Encountered issue with setting the description in the Svelte CSF format, above the export const meta
>
> Could you elaborate on this?

What I recall from refactoring the parser in `@storybook/addon-svelte-csf`, there was a small issue with not being able to read the leading comment while using `lang=ts`. I  don't remember the exact reason of what caused it, but I think it was execution order with preprocess at compile. I can take a look at it tomorrow.

**Pull request merged**